machineconfiguration/v1/types_machineconfignode.go

```diff
@@ -98,14 +100,23 @@ type MachineConfigNodeSpec struct {
 	// the new machine config against the current machine config.
 	// +required
 	ConfigVersion MachineConfigNodeSpecMachineConfigVersion `json:"configVersion"`
+
+	// configImage holds the desired image for the node targeted by this machine config node resource.
+	// The desired image represents the image the node will attempt to update to and gets set before the machine config operator validates
+	// the new image against the current image.
+	// +openshift:enable:FeatureGate=ImageModeStatusReporting
+	// +optional
+	ConfigImage MachineConfigNodeSpecConfigImage `json:"configImage"`
 }
 
 // MachineConfigNodeStatus holds the reported information on a particular machine config node.
 type MachineConfigNodeStatus struct {
 	// conditions represent the observations of a machine config node's current state. Valid types are:
 	// UpdatePrepared, UpdateExecuted, UpdatePostActionComplete, UpdateComplete, Updated, Resumed,
-	// Drained, AppliedFilesAndOS, Cordoned, Uncordoned, RebootedNode, NodeDegraded, PinnedImageSetsProgressing,
-	// and PinnedImageSetsDegraded.
+	// Drained, AppliedFilesAndOS, AppliedOSImage, AppliedFiles, Cordoned, Uncordoned, RebootedNode, NodeDegraded, PinnedImageSetsProgressing,
+	// ImagePulledFromRegistry, and PinnedImageSetsDegraded.
+	// The following types are only available when the ImageModeStatusReporting feature gate is enabled: ImagePulledFromRegistry,
+	// AppliedOSImage, AppliedFiles
```

Comment:
I would remove these from the mentioned list above if you are going to mention these only exist in the conditions under specific circumstances.